index.js

```diff
@@ -3983,6 +3983,31 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Type-checked version of String#split that accepts RegEx as a separator.
+  //.
+  //. Takes a pattern and a string, and returns the result of splitting the
+  //. string at every non-overlapping occurrence of the pattern (possibly
+  //. removing the matched substrings, depending on a pattern).
+  //.
+  //. [Full documentation on MDN.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split)
+  //.
+  //. See also [`splitOn`](#splitOn).
+  //.
+  //. ```javascript
+  //. > S.splitOnRegex(/[,;][ ]*/, 'foo, bar, baz')
+  //. ['foo', 'bar', 'baz']
+  //.
+  //. > S.splitOnRegex(/[,;][ ]*/, 'foo;bar;baz')
+  //. ['foo', 'bar', 'baz']
+  //. ```
+  S.splitOnRegex =
+  def('splitOnRegex',
+    {},
+    [$.RegExp, $.String, $.Array($.String)],
+    splitOn);
```

Comment:
<del>Please indent the three lines above so that all four arguments to `def` are aligned.</del>

<ins>Please place all four arguments to `def` on the same line (line wrapping is not necessary in this case).</ins>